Large scale data

jorainer opened the pull request:
Add the pre-computed large scale vignette. While building the package works nicely, pkgdown does not seem to work, because it can't find the image...

jorainer commented on vignettes/large_scale_analysis.Rmd:
````diff
@@ -0,0 +1,1003 @@
+``` r
+kc <- read_xlsx("vignettes/known_compounds_largescale.xlsx") |>
+    as.data.frame()
+#> Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': `path` does not exist: 'vignettes/known_compounds_largescale.xlsx'
````
this is very odd... the file **is** in that folder and rendering was done in the Metabonaut base folder. so, the file is in vignettes/...

philouail replied:
but, i thought when you render it always use as base path the path where the rmd file is. in that logic it make sense is failing no ?

philouail replied:
same if you manually run the rmd, the working directory is the rmd file path, so this would fail too no ?

jorainer replied:
I defined the compounds manually in the qmd and removed the xlsx sheet. makes also the vignette less dependent of external files.

jorainer commented:
@philouail can you try with the new file? I ran the knitting in the vignettes folder and save the figures also there... seems to work for me now...

philouail commented:
@jorainer so I changed some stuff around to make it website-ready

philouail reviewed (approved):
The tests are passing, didn't see anything wrong with the code, so it's all good on my side.

jorainer commented:
Great! Thanks for the updates! but please don't merge yet - I did not yet render it completely.

jorainer commented:
This is now the *full* vignette. Maybe we need to fix/adapt some text descriptions or typos here and there. Also, I need to maybe add some more information on the computer on which this was rendered.

philouail commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+p <- peakRAM(
+    twins <- refineChromPeaks(twins, param = mnpp, chunkSize = 8)
+)
+
```
extra

philouail commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+data object representing the MS data of the experiment. Note that this size
+could be further reduced (and eventually overall performance improved) by
+storing the MS data in a SQL database and representing/interfacing it with a
+`Spectra` backend from the `r Biocpkg("MsBackendSql")` package.
```
wait but isn't that what you do at the beginning ?

jorainer replied:
Nope. At the beginning it was some left-over from a previous analysis. Now the full analysis is run using the `MsBackendMzR`. I moved the code from above down here.

philouail commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+We perform chromatographic peak detection on this subset for different
+configurations:
+
+- Peak detection using the default `XcmsExperiment` result object.
```
this whole intro part is very clear i like it a lot

philouail commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+#' Total RAM used: 50MB for XcmsExperiment, 0MB for others
+#' Peak RAM used: 7.5GB for all
+#' Time: ~ 2000, 1200, 750, 600 seconds with about 5-10 seconds more for HDF5
+
```
extra line here and below

philouail commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+peak RAM usage for `chunkSize = 8` is, for the present data set, about twice as
+large as the size of the respective MS data in memory. The processing time is
+reduced with an increase in the number of CPUs used, but this relationship is
+not linear.
```
Maybe i missed it, but do we want to say explain why it is not linear ? This is  a very technical vignette so i think it would have its place here.

jorainer replied:
Hm, OK. I try to explain - although it's also not very clear to me :)

philouail commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+
+
+## Details of the software structure and data flow
+
```
No comment on the paragraph below i think it is very clear, I would however maybe separate it into smaller paragraphs just to improve the reading ?

philouail commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+  since parallel processing involves also distribution of data to, and
+  collection of results from, the individual processing nodes.
+- For the present analysis, a big part of computation time was spent on data
+  import from the raw data files. CDF files might not be the most efficient data
```
I would re-precise here, "import the raw data files from MetaboLights" aha

jorainer replied:
Ah, no, sorry, I don't mean this import - but the on-the-fly retrieval of MS data. I will see how I can rephrase

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
````diff
@@ -0,0 +1,1015 @@
+
+table(sampleData(twins)$sex)
+```
+
````
Maybe better I mention this at the very last section to not confuse users here.

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+## Parallel processing setup
+
+Many functions from *xcms*, in particular the ones requiring heavy calculations,
+support parallel processing. Processing functions, will then split the data
```
thanks! will fix

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+processing also on regular computer setups. In this document a large public
+metabolomics data set is analyzed and *xcms*' memory usage and performance
+tracked. Performance and memory usage for different configurations are compared
+on a smaller data subset. Finally, details on the data flow in *xcms* are
```
True, it's not clear. I rephrase that.

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+*xcms*, which include full support of the MS data infrastructure provided by the
+*Spectra* package as well as a new *on-disk* storage mode for preprocessing
+results, further reduce *xcms*' memory demand and hence enable large scale data
+processing also on regular computer setups. In this document a large public
```
Yes, thanks!

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+
+# Introduction
+
+By design, *xcms* supports preprocessing of large scale data. Recent updates to
```
Indeed! thanks, I will change and specify

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+**Note**: this vignette is
+[**pre-computed**](https://ropensci.org/blog/2019/12/08/precompute-vignettes/).
+See the session info for information on packages used and the date the vignette
+was rendered.
```
Yes. I will expand this section.

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+Many functions from *xcms*, in particular the ones requiring heavy calculations,
+support parallel processing. Processing functions, will then split the data
+among these processes and perform the calculations in parallel. While parallel
+processing can reduce the processing, it is important to note that it also
```
yes, thanks!

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+among these processes and perform the calculations in parallel. While parallel
+processing can reduce the processing, it is important to note that it also
+requires all data that is being processed to be in memory. There should thus
+always be a balance between the number of parallel processes and the available
```
thanks, yes, will change

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+Based on the BPC above, we filter the data set to spectra measured between 20
+and 900 seconds. Such restriction of the data set also avoids to perform the
+chromatographic peak detection in the part of the LC where no compounds are
+expected to elute.
```
Makes sense, yes.

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+    caption = paste0("Peak RAM memory usage and processing time for ",
+                     "the chromatographic peak detection step on the",
+                     " full data."))
+
```
you don't like the extra lines ;)

jorainer commented on vignettes/large-scale-analysis.Rmd.orig:
```diff
@@ -0,0 +1,1015 @@
+determine also their total sum. Here we take advantage of the possibility to
+load only selected columns of the chromatographic peak matrix, which will reduce
+the memory need for the present calculation. Also, we use `bySample = TRUE`
+which returns the result as a `list` of chromatographic peak matrices, one per
```
one list with `length` being equal to the number of samples. I will rephrase.

jorainer commented:
I addressed all your comments @philouail - let me know if there are new/additional things to change. Otherwise I would run render the vignette locally and then also add the Rmd - just to be sure we do that before merging ;)

jorainer commented:
@philouail , all good from my side. Please have a look and let me know if there's something else I need to change.
